python/feathub/online_stores/tests/test_mysql_online_store.py

```diff
@@ -0,0 +1,188 @@
+
+
+# TODO: Restructure these test cases in a way similar to ProcessorTestBase.
+# TODO: move this class to python/feathub/online_stores/tests folder after
+#  the resource leak problem of other flink processor's tests is fixed.
+class MySQLOnlineStoreITTest(ABC, FeathubITTestBase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.setup_mysql_container()
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        cls.teardown_mysql_container()
+
+    def setUp(self) -> None:
+        super().setUp()
+        input_data = pd.DataFrame(
+            [
+                [1, 1, datetime(2022, 1, 1, 0, 0, 0).strftime("%Y-%m-%d %H:%M:%S")],
+                [2, 2, datetime(2022, 1, 1, 0, 0, 1).strftime("%Y-%m-%d %H:%M:%S")],
+                [3, 3, datetime(2022, 1, 1, 0, 0, 2).strftime("%Y-%m-%d %H:%M:%S")],
+            ],
+            columns=["id", "val", "ts"],
+        )
+
+        self.feature_table_schema = (
+            Schema.new_builder()
+            .column("id", types.Int64)
+            .column("val", types.Int64)
+            .column("ts", types.String)
+            .build()
+        )
+
+        file_source = self.create_file_source(
+            input_data,
+            keys=["id"],
+            schema=self.feature_table_schema,
+            timestamp_field="ts",
+        )
+        self.table_name = "feature_table"
+        sink = MySQLSink(
+            host=self.mysql_container.get_container_host_ip(),
+            port=int(
+                self.mysql_container.get_exposed_port(
+                    self.mysql_container.port_to_expose
+                )
+            ),
+            database=self.mysql_container.MYSQL_DATABASE,
+            username=self.mysql_container.MYSQL_USER,
+            password=self.mysql_container.MYSQL_PASSWORD,
+            table=self.table_name,
+        )
+
+        engine = sqlalchemy.create_engine(self.mysql_container.get_connection_url())
+        with engine.begin() as conn:
+            conn.execute(
+                sqlalchemy.text(
+                    f"CREATE TABLE {self.table_name} ("
+                    f"id BIGINT PRIMARY KEY, val BIGINT, ts TEXT)"
+                )
+            )
+
+        self.client.materialize_features(file_source, sink, allow_overwrite=True).wait(
+            30_000
+        )
+
+        self.on_demand_feature_view = OnDemandFeatureView(
+            name="on_demand_feature_view",
+            features=[
+                "feature_table.val",
+            ],
+            keep_source_fields=True,
+            request_schema=Schema.new_builder().column("id", types.Int64).build(),
+        )
+
+    def tearDown(self) -> None:
+        super().tearDown()
+        engine = sqlalchemy.create_engine(self.mysql_container.get_connection_url())
+        with engine.begin() as conn:
+            conn.execute(sqlalchemy.text(f"DROP TABLE IF EXISTS {self.table_name}"))
+
+    def test_get_online_features(self):
```

Comment:
We implemented mysql client rather than mysql online store in FeatHub. This test is actually testing `MySQLSource`.

Would it be more intuitive to move this test to `MySQLSourceSinkITTest`?